### PR TITLE
Ft manage donations

### PR DESCRIPTION
**Pull request by tasiukwaplong**
#### What does this PR do?
Making donations to the farmer. This feature enables donations be made to farmers by means of making payment via paystack.

#### Description of Task to be completed?
list the things you did to complete this project
- Setting up of paystack account for collection of payments  
- Setting up webhook using keys provided by paystack

#### How should this be manually tested?
- Clone this repo 
- Run `npm install` to get all necessary packages
- Create a database on postgress
- Use .env.sample to fill the connection details to postgres database you created and rename to `.env`
- Still in the `.env` file, input the secret key you got from (paystack.com)[paystack.com]
- Run `npm run db:migrate` to run all migrations
- Deploy online and serve using HTTPS 

#### Any background context you want to provide?
No

#### What are the relevant pivotal tracker stories?(if applicable)
None  

#### Screenshots
Paystack test payment: https://paystack.com/pay/yopiuop6hl  
[UPDATED] No userid anymore  
Paystack

Successful payment
pic2

Webhook is sent to this url: (http://nmwrrd.000webhostapp.com/u/log.txt) [UPDATED] No userid anymore[http://nmwrrd.000webhostapp.com/u/log.txt] to see its output

pic3

### CODE REVIEW

File: src/modules/auth/auth.controller.js [hunk 1]
```diff
@@ -19,4 +22,25 @@ export const createUser = async (req, res) => {
   }
 };
 
+export const makeDonation = async (req, res) => {
```
Comment (teezyfortune) [inline src/modules/auth/auth.controller.js]: Nice work, however the make donation is meant  to be in its own folder called donation, then create a donation.controller.js inside it

File: src/modules/auth/auth.route.js [hunk 2]
```diff
@@ -1,11 +1,13 @@
 import express from 'express';
-import { createUser } from './auth.controller';
+import { createUser, makeDonation } from './auth.controller';
```
Comment (teezyfortune) [inline src/modules/auth/auth.route.js]: well done and good job!, Normally makeDonation function should be in its own routes file inside the donation files you created in side modules, please can you remove the make donation from auth controller and import it seperately after you have created its own route file, thanks

File: src/modules/auth/auth.controller.js [hunk 3]
```diff
@@ -19,4 +22,25 @@ export const createUser = async (req, res) => {
...
+  // this will be where webhook call will hit
```
Comment (teezyfortune) [inline src/modules/auth/auth.controller.js]: well done and nice implementation, actually i dont understand what you are doing here, are you calling paystack directly from payment. anyways I  would know why and would like to understand and maybe you run a demo for us

File: src/modules/auth/auth.controller.js [hunk 4]
```diff
@@ -19,4 +22,25 @@ export const createUser = async (req, res) => {
...
+  // testing this is hwoever subject to if this project on a server serving on HTTPS
```
Comment (teezyfortune) [inline src/modules/auth/auth.controller.js]: Hello, also please kindly remove any of comment you know are not relevant to your implementation, you can leave the ones explaining you implementations though

File: src/services/users/donations.services.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import models from '../../database/models';
```
Comment (teezyfortune) [inline src/services/users/donations.services.js:1]: Nice work,  Please try and follow the convention, no '.**then**', use async await  and try catch block, follow the ways others are written, well done

**Review by koladeore — DISMISSED** [review]

**Review by teezyfortune — DISMISSED** [review]

**Review by ucheuzor — DISMISSED** [review]

File: src/modules/auth/auth.route.js (hunk 2 above)
Comment (tasiukwaplong) [reply src/modules/auth/auth.route.js]: done

**Review by koladeore — APPROVED** [review]

**Review by ucheuzor — APPROVED** [review]

**Review by MrDannie — APPROVED** [review]

**Review by teezyfortune — APPROVED** [review]

**Pull request merged**